### DISCUSSION TITLE
RoleDoesNotExist: guard_name defaults to 'sanctum' when seeding in multitenancy setup

### DISCUSSION THREAD

**Discussion by michaelhume** [spatie/laravel-permission#2040] [Q&A] [answered]
Hi all, 

I'm playing with some different setups for roles & perms in a multi-tenancy setup - saw #1744 and may investigate further, but my case will likely require tenants to modify some roles/perms.

When seeding my roles/perms during the tenant setup process, the guard_name is being set to 'sanctum' - resulting in a `RoleDoesNotExist` exception when I attempt to assign the role. 

I am confused about where this is coming from - I don't even have sanctum listed as a possible guard. I can solve this by hardcoding 'web' as the guard_name during creation, but I'd like to understand what is happening here. 

I am using the multi-tenancy package, and  extending the Role & Perm model to add `use UsesTenantConnection;` trait. My User model defines `$guard_name = 'web'`

I am running the seed command in a controller 

[Controller.php]
``` php

$tenant->makeCurrent();
...
...

// seed roles and permissions
Artisan::call("tenants:artisan 'db:seed --class RolesAndPermissionsSeeder' --tenant=".$tenant->id);

$admin->assignRole('super-admin');   // <----- exception thrown here

Tenant::forgetCurrent();

```

Some other relevant bits:

[auth.php]
```php

'defaults' => [
        'guard' => 'web',
        'passwords' => 'users',
    ],

'guards' => [
        'web' => [
            'driver' => 'session',
            'provider' => 'users',
        ],
        'landlord' => [
            'driver' => 'session',
            'provider' => 'landlords',
        ],
],

```

[sanctum.php]
```php

'guard' => 'web'

```

When I run this in tinker, or console I get the correct guard. I suspect this has to do with a knowledge gap in sanctum, since even doing `Permission::create(['guard_name'=> config('auth.defaults.guard'), 'name' => $permission]);` in the seeder produces the same issue. Is sanctum updating the default guard somewhere??

Any thoughts or advice greatly appreciated!

**Comment by angeljqv** [comment]
>Is sanctum updating the default guard somewhere??

Yes, but on your implementation

**Reply by michaelhume** [reply]
True - I do manipulate the guards when switching between "landlord" and "tenant" context/guards - but I don't modify the `auth.defaults.guard`. In sanctum - I define the guard as `web` as well.

It's not clear to me where the `sanctum` guard is coming from.

**Reply by michaelhume** [reply]
Meaning the actual string for `guard_name` is set to sanctum.

**Reply by angeljqv** [reply]
https://github.com/spatie/laravel-permission/blob/main/src/Guard.php
That is the class which search the guard_name, you can debug that for find where is the updating

**Comment by michaelhume** [comment, marked as answer]
For the record, this has nothing to do with the package at all - it was in the route definition at the top of the file:

```php

Route::middleware(['auth:sanctum', 'verified'])->group(.....

```

PEBKAC.
